docs/error.md

```diff
@@ -0,0 +1,60 @@
+# Error
+
+An object representing an error that can be triggered / thrown.
```

Comment:
An error is defined as an object that represents an error that can be triggered or thrown.